keras_hub/src/models/parseq/parseq_tokenizer.py

```diff
@@ -0,0 +1,140 @@
+import re
+
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.tokenizers import tokenizer
+from keras_hub.src.utils.tensor_utils import is_int_dtype
+from keras_hub.src.utils.tensor_utils import is_string_dtype
+from keras_hub.src.utils.tensor_utils import preprocessing_function
+
+PARSEQ_VOCAB = (
+    "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"
+    "\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+)
+
+try:
+    import tensorflow as tf
+except ImportError:
+    tf = None
+
+
+@keras_hub_export(
+    [
+        "keras_hub.tokenizers.PARSeqTokenizer",
+        "keras_hub.models.PARSeqTokenizer",
+    ]
+)
+class PARSeqTokenizer(tokenizer.Tokenizer):
```

Comment:
Hi @mattdangerw,

When you get some time can you take look at this tokenizer, the original implementation is here: 

Preprocess for dataset: https://github.com/baudm/parseq/blob/main/strhub/data/dataset.py
Tokenizer: https://github.com/baudm/parseq/blob/main/strhub/data/utils.py#L102

I followed roughly other tokenizers in kerashub and wrote this, we can further discussion on what changes are required.

Thanks!